Anoosha-foodon

Anoosha-Sehar opened the pull request:
-Replaced the deprecated terms with non-deprecated ones.
-Moved axiom (where applicable) to src/ontology/imports/foodon_product_import.owl
-Changed the relation (where needed)

Anoosha-Sehar committed:
Updated 'cow lung (whole, raw)'

Replaced the deprecated term.

Anoosha-Sehar committed:
Updated 'beef food product'

Replaced the deprecated term.

Anoosha-Sehar committed:
Updated 'cattle sausage food product'

-Replaced the deprecated term

Anoosha-Sehar committed:
Updated 'luncheon meat'

-Replaced the deprecated term.

Anoosha-Sehar committed:
Updated 'blood meal'

-Replaced the deprecated term.

Anoosha-Sehar committed:
Updated 'beef fat'

-Replaced the deprecated term
-Moved the axiom to 'src/ontology/foodon_product_import.owl

Anoosha-Sehar committed:
Updated 'cow blood'

-Replaced the deprecated term

Anoosha-Sehar committed:
Updated 'beef bacon'

-Replaced the deprecated term
-Changed the relation from 'derives to' to 'has defining ingredient'
-Moved the axiom to 'src/ontology/foodon_product_import.owl

Anoosha-Sehar committed:
Updated 'beef cut (cured)

-Replaced the deprecated term.
-changed the relation.

Anoosha-Sehar committed:
Updated 'beef (cured, cooked)'

-Replaced the deprecated term
-Changed the relation from 'derives to' to 'has defining ingredient'
-Moved the axiom to 'src/ontology/foodon_product_import.owl

Anoosha-Sehar committed:
Updated 'salami'

-Replaced the deprecated term

Anoosha-Sehar committed:
Updated 'beef gravy'

-Replaced the deprecated term
-Moved the axiom to 'src/ontology/foodon_product_import.owl

Anoosha-Sehar committed:
Updated 'beef gravy mix'

-Replaced the deprecated term
-Moved the axiom to 'src/ontology/foodon_product_import.owl

Anoosha-Sehar commented:
Special attention can be given to multi-ingredient/multi component terms like beef gravy, beef (cured, cooked), beef bacon etc.

ddooley commented on src/ontology/imports/foodon_product_import.owl:
```diff
@@ -62190,6 +62196,12 @@ soy grits have a particle size larger than 100 mesh.&quot; [https://link.springe
+        <rdfs:subClassOf>
+            <owl:Restriction>
+                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/RO_0001000"/>
+                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/FOODON_03302091"/>
```
Probably better to say this derives from some bos taurus.  Saying it derives from beef broth seems a bit circular.

ddooley replied:
actually just take this out, and leave it be handled by parent class.

Anoosha-Sehar replied:
Right, 'beef broth was added as comment section also support this statement by saying that it * derives from 'broth or stock'. But I agree that it seems a bit circular logic.
Secondly, it has a parent class 'cattle broth or stock food product', which itself should 'derives from' some 'bovinae'. Thus I think'beef' in 'beef gravy' can only be defined if we add axiom in this class ('derives from' some 'Bos taurus'). OBecause otherwise it cannot be handled by its parent class.
Same goes for 'beef gravy mix'.

Anoosha-Sehar replied:
Okay I've checked. 'Bos taurus', 'Bos primigenius taurus', 'cattle' are all synonyms as per NCBI.
We can add the axiom in parent term which is 'cattle broth or stock food product' so I have updated 'cattle broth or stock food product' and 'cattle cured meat food product'.

ddooley commented on src/ontology/imports/foodon_product_import.owl:
```diff
@@ -136151,6 +136163,12 @@ sidol</oboInOwl:hasNarrowSynonym>
 
     <owl:Class rdf:about="http://purl.obolibrary.org/obo/FOODON_03310598">
         <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/FOODON_00001695"/>
+        <rdfs:subClassOf>
```
As above, remove this, and put derives from bos taurus axiom on parent instead.

ddooley commented on src/ontology/imports/foodon_product_import.owl:
```diff
@@ -148852,6 +148870,12 @@ size larger than 100 mesh, U.S. Standard Sieve Size.</obo:IAO_0000115>
 
     <owl:Class rdf:about="http://purl.obolibrary.org/obo/FOODON_03311654">
         <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/FOODON_00001696"/>
+        <rdfs:subClassOf>
```
Delete this and instead put derives from bos taurus on the 'beef food product' parent.  All children of 'beef food product' don't need a derives from clause any more then.

Anoosha-Sehar replied:
Okay done

ddooley reviewed (changes requested):
Mainly, lets have 'beef food product' have a 'derives from' some Bos Taurus axiom; and drop all the subordinate class 'derives from' axioms.

Anoosha-Sehar committed:
Deleted axiom from 'beef bacon', beef (cured, cooked), beef cut (cured)

Anoosha-Sehar committed:
Updated 'beef food product'

Anoosha-Sehar committed:
Updated 'cattle cured food product'

Added an axiom 'derives from' some 'Bos taurus'

Anoosha-Sehar committed:
Updated 'cattle broth or stock food product'

Added an axiom 'derives from' some 'Bos taurus'

Anoosha-Sehar committed:
Deleted 'derives from' axiom from 'beef gravy' and 'beef gravy mix'

Anoosha-Sehar commented:
Summary of all changes that have been made:
-Dropped the axiom 'derives from' from 'beef bacon', beef (cured, cooked), beef (cut, cured) and instead added in a parent class 'beef food product' and 'cattle cured meat food product'.
-Dropped the axiom 'derives from' from 'beef gravy' and 'beef gravy mix' and instead added in parent class 'cattle broth or stock food product'.
